nyc_taxis/challenges/indexing-querying.json

```diff
@@ -0,0 +1,131 @@
+{
+  "name": "indexing-querying",
+  "description": "Indexes half the data for NYC Taxis, then concurrently queries Elasticsearch while indexing the remaining data.",
+  "default": false,
+  "schedule": [
+    {
+      "operation": "delete-index"
+    },
+    {
+      "operation": {
+        "operation-type": "create-index",
+        "settings": {%- if index_settings is defined %} {{index_settings | tojson}} {%- else %} {
+        "index.codec": "best_compression",
+        "index.refresh_interval": "30s",
+        "index.translog.flush_threshold_size": "4g"
+      }{%- endif %}
+    }
+  },
+    {
+      "name": "check-cluster-health",
+      "operation": {
+        "operation-type": "cluster-health",
+        "index": "nyc_taxis",
+        "request-params": {
+          "wait_for_status": "{{cluster_health | default('green')}}",
+          "wait_for_no_relocating_shards": "true"
+        },
+        "retry-until-success": true
+      }
+    },
+    {
+      "name": "initial-bulk",
+      "clients": {{bulk_indexing_clients | default(8)}},
+      "operation": {
+        "operation-type": "bulk",
+        "bulk-size": {{bulk_size | default(10000)}},
+        "ingest-percentage": 50
+      }
+    },
+    {
+      "name": "refresh-after-index",
+      "operation": "refresh"
+    },
+    {
+      "operation": {
+        "operation-type": "force-merge",
+        "request-timeout": 7200{%- if force_merge_max_num_segments is defined %},
+    "max-num-segments": {{ force_merge_max_num_segments | tojson }}
+    {%- endif %}
+  }
+  },
+    {
+      "name": "refresh-after-force-merge",
+      "operation": "refresh"
+    },
+    {
+      "operation": "wait-until-merges-finish"
+    },
+    {
+      "parallel": {
+        "completed-by": "default",
+        "tasks": [
+          {
+            "name": "concurrent-bulk",
+            "operation": {
+              "operation-type": "bulk",
+              "bulk-size": {{bulk_size | default(10000)}},
+              "ingest-percentage": 50
```

Comment:
Is the intention to run in throughput mode (i.e. at full indexing speed, rather than limit it?) while queries also run simultaneously?

Reply:
we are measuring hits to indexing throughput as well as query latency, so yes